src/main/java/io/deephaven/csv/reading/CsvReader.java

```diff
@@ -195,8 +197,26 @@ private static Result commonReadLogic(final CsvSpecs specs, CellGrabber grabber,
                     for (Future<Object> readerFuture : readerFutures) {
                         readerResults.add((ParseDenseStorageToColumn.Result) readerFuture.get());
                     }
+                } catch (Throwable t) {
+                    innerThrowable = t;
                 } finally {
                     executorService.shutdownNow();
+                    final boolean allTerminated =
+                            executorService.awaitTermination(specs.threadShutdownTimeout(), TimeUnit.MILLISECONDS);
+                    if (!allTerminated) {
+                        final String message = String.format("Failed to shutdown all threads (Waited %d milliseconds)",
+                                specs.threadShutdownTimeout());
+
+                        if (innerThrowable != null) {
+                            throw new RuntimeException(message, innerThrowable);
+                        } else {
+                            throw new RuntimeException(message);
+                        }
+                    }
+
+                    if (innerThrowable != null) {
+                        throw innerThrowable;
+                    }
```

Comment:
We want to show `innerThrowable` to the user, not our timeout. Also, I kind of hate bare `RuntimeException`. How about:
```suggestion
                    if (!allTerminated) {
                        final String message = String.format(
                                "Failed to shutdown all threads (after waiting %d milliseconds)",
                                specs.threadShutdownTimeout());
                        if (innerThrowable == null) {
                            throw new TimeoutException(message);
                        }
                        innerThrowable.addSuppressed(new TimeoutException(message));
                    }
```